agent/pkg/chassis/loadbalancer/consistenthash/consistenthash.go

```diff
@@ -34,7 +34,9 @@ type Strategy struct {
 // ReceiveData receive data.
 func (s *Strategy) ReceiveData(inv *invocation.Invocation,
 	instances []*registry.MicroServiceInstance, serviceName string) {
+	s.mtx.Lock()
 	s.instances = instances
+	s.mtx.Unlock()
```

Comment:
why here need a lock?

Reply:
slice `s.instances` for-range operation has lock to protect, and so i think assignment also needs a lock

Reply:
for range s.instances and return s.instances[index] has lock to protect is ture. but i think assignment doesn't needs a lock. @Poorunga  what's your opinion?

Reply:
We need to judge whether this is a critical section, if it belongs to a critical section in the case of concurrent execution, then we must lock it.

Reply:
ok, agree